app/Http/Controllers/Twill/BaseApiController.php

```diff
@@ -80,37 +122,231 @@ protected function getBrowserTableData(\Illuminate\Support\Collection|\Illuminat
         return $results;
     }
 
-    protected function getApiRepository()
+    public function getIndexItems(array $scopes = [], bool $forcePagination = false): Collection|LengthAwarePaginator
     {
-        return $this->app->make("{$this->namespace}\Repositories\\Api\\" . $this->modelName . 'Repository');
+        if (TwillPermissions::enabled() && TwillPermissions::getPermissionModule($this->moduleName)) {
+            $scopes += ['accessible' => true];
+        }
+
+        $requestFilters = $this->getRequestFilters();
+        $appliedFilters = [];
+        $this->applyQuickFilters($requestFilters, $appliedFilters);
+        $this->applyBasicFilters($requestFilters, $appliedFilters);
+        return $this->transformIndexItems(
+            $this->getApiData($scopes, $forcePagination, $appliedFilters)
+        );
     }
 
-    protected function getIndexItems($scopes = [], $forcePagination = false)
+    /**
+     * Get the applied quick filter.
+     */
+    protected function applyQuickFilters(array &$requestFilters, array &$appliedFilters): void
     {
-        $perPage = request('offset') ?? $this->perPage ?? 50;
-        $items = $this->getApiRepository()->get($this->indexWith, $scopes, $this->orderScope(), $perPage, $forcePagination);
+        if (array_key_exists('status', $requestFilters)) {
+            $filter = $this->quickFilters()->filter(
+                fn (QuickFilter $filter) => $filter->getQueryString() === $requestFilters['status']
+            )->first();
+
+            if ($filter !== null) {
+                $appliedFilters[] = $filter;
+            }
+        }
+        unset($requestFilters['status']);
+    }
+
+    /**
+     * Get other filters that need to applied. Use the API search filter when
+     * requested.
+     */
+    protected function applyBasicFilters(array &$requestFilters, array &$appliedFilters): void
+    {
+        foreach ($requestFilters as $filterKey => $filterValue) {
+            $filter = $this->filters()->filter(
+                fn (BasicFilter $filter) => $filter->getQueryString() === $filterKey
+            )->first();
+
+            if ($filter !== null) {
+                $appliedFilters[] = $filter->withFilterValue($filterValue);
+            } elseif ($filterKey === 'search') {
+                $appliedFilters[] = Search::make()
+                    ->searchFor($filterValue)
+                    ->searchColumns($this->searchColumns);
+            }
+        }
+    }
+
+    public function getApiData($scopes = [], $forcePagination = false, $appliedFilters = [])
+    {
+        return $this->getApiRepository()->get(
+            with: $this->indexWith,
+            scopes: $scopes,
+            orders: $this->orderScope(),
+            perPage: $this->request->get('offset') ?? $this->perPage,
+            forcePagination: $forcePagination,
+            appliedFilters: $appliedFilters
+        );
+    }
+
+    protected function getIndexTableColumns(): TableColumns
```

Comment:
This is mostly a copy of Twill's `ModuleController::getIndexTableColumns()` but with a few datahub-specific default columns added.